logstash-core/lib/logstash/config/source_loader.rb

```diff
@@ -0,0 +1,68 @@
+# encoding: utf-8
+require "logstash/config/source/local"
+require "logstash/errors"
+require "thread"
+require "set"
+
+module LogStash module Config
+  class SourceLoader
+    class Loader
+      def initialize(loaders)
+        @loaders = loaders
+      end
+
+      def fetch
+        @loaders.collect(&:pipeline_configs).flatten
+      end
+    end
+
+    include LogStash::Util::Loggable
+
+    def initialize
+      @source_loaders_mutex = Mutex.new
+      @source_loaders = Set.new([LogStash::Config::Source::Local])
```

Comment:
the current class is called SourceLoader, but this variable is named `source_loaders` and assign to it a set of Source classes